### ISSUE TITLE
Better define the package's scope

### ISSUE THREAD

**Issue by niksirbi** [neuroinformatics-unit/movement#59]
- Remove the "term" kinematic" from the subtitle (it may be too restrictive)
- Describe what kind of data we deal with
- Better describe what sort of analyses we want to enable

**Comment by talmo** [comment]
It'd be great to learn a bit about your guys' roadmap. We routinely have a need for downstream analyses, and we don't have a great place to stick our egocentrization, normalization, smoothing, visualization, etc code.

If it's a good fit for the scope, we'd be happy to shoot some PRs your way!

Here's some examples of our use cases:

- [ROI definition, geometric feature extraction, visualization and movie rendering](https://gist.github.com/talmo/e90f73c5838a4ab886a7e5cae94c4aa1)
- [Large-scale (53M frames) exploratory data analysis](https://gist.github.com/talmo/d21b4ba170e109094159401ca3338bdc)
- [Gait feature extraction](https://drive.google.com/file/d/1yz0LSbWI0eg7-kHcyz_EZ5-Wf8umXUqe/view?usp=sharing)

Let me know if any of this is within scope!

Cheers,

Talmo

**Comment by niksirbi** [comment]
Hey @talmo,

as this issue showcases, we were just about discussing our scope and what to include in there.
In short, almost everything you mentioned is in our plans, including (but not limited to):

- **preprocessing of pose tracks**
	- handling missing values, interpolation
	- smoothing, resampling, filtering
- **visualisation**, incl. (at minimum):
	- trajectory plots
	- occupancy heatmaps
- ROI support:
	- defining, saving, and loading ROIs
	- metrics related to ROIs (time spent in, entries/exits, etc.)
- Coordinates systems and transforms, e.g.
	- egocentric transform
	- aligning multiple videos to a common coordinate space for group analysis
- Simple kinematic calculations
	- velocity and acceleration
	- joint angles
- Maybe modules for specialised applications, e.g.:
	- pupilometry
	- gait analysis
	
What is **NOT** in our scope is anything that would rely on GPU-heavy ML. We want to keep this package simple and accessible, aiming for students to be able to run it on their laptops (so we would like to avoid tensorflow, pytorch).

Any other computation that can be done on pose tracks can be brought into the roadmap (depending on how many people ask for it). Our immediate goal is to serve the needs of labs at the [Sainsbury Wellcome Centre](https://www.sainsburywellcome.org/web/). That said, because the behavioural projects running here are extremely diverse, our scope is broad and flexible. You can check out [this presentation](https://neuroinformatics-unit.github.io/behavioural-projects-slides/#/title-slide) for an overview of our projects (slides 16-24 are about `movement`).

You are also very welcome to chat with us on [our zulip chat](https://neuroinformatics.zulipchat.com/#narrow/stream/406001-Movement). We started it just now but it's meant for discussions just as this one. 

I'm also open to chatting online some time in the following weeks. The project is still at an early stage, so we would be very open to taking your needs into scope.

**Comment by niksirbi** [comment]
Hey @talmo, our docs now include a [community section](https://movement.neuroinformatics.dev/community/index.html#), which contains the project's mission, scope and roadmap (both short- and long-term). I hope this will clarify the scope question and will help to guide our potential collaborations with you, and with others.

Don't hesitate to [get in touch](https://neuroinformatics.zulipchat.com/#narrow/stream/406001-Movement) if the docs are unclear or if you think additional features should be within our roadmap.
